### PR TITLE
Check addressbook permission before moving

### LINKED ISSUE #174: contact is deleted after move to read only address book
### Steps to reproduce
1. User has an contact in his own address book
2. An other user share an address book with this user in read only mode
3. User move an contact to the shared address book

### Expected behaviour
1. the shared read only address book is not selectable or
2. you get an valid error message

### Actual behaviour
1. error message: "contact is not createable"
2. the origin contact is deleted

### Server configuration
**Operating system**:
rasperian jessy

**Web server:**
Apache

**Database:**
mysql 5.5.54

**PHP version:**
5.6.29

**Nextcloud version:** (see Nextcloud admin page)
11.0.2

**Contacts version:** (see Nextcloud apps page)
1.5.3

**Updated from an older Nextcloud or fresh install:**
updated
**Signing status:**
No errors have been found.

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "ocb41c453080",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "\/var\/local\/owncloud\/data",
        "dbtype": "mysql",
        "version": "11.0.2.7",
        "installed": true,
        "forcessl": false,
        "theme": "",
        "maintenance": false,
        "default_language": "de",
        "enable_previews": true,
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "dbname": "nextcloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "owncloud",
        "mail_smtpmode": "smtp",
        "mail_domain": "cipicek.click",
        "mail_smtpsecure": "tls",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "posteo.de",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "loglevel": 1,
        "log_authfailip": true,
        "logfile": "\/var\/log\/owncloud\/owncloud.log",
        "log_rotate_size": 2097152,
        "logtimezone": "Europe\/Berlin",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trashbin_retention_obligation": "auto",
        "updatechecker": false,
        "appstoreenabled": true,
        "appstoreurl": "https:\/\/api.owncloud.com\/v1",
        "htaccess.RewriteBase": "\/",
        "appstore.experimental.enabled": true
    }
}

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
local

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

### Client configuration
**Browser:**
firefox

**Operating system:**
debian strech

**CardDAV-clients:**
Web interface

### Logs
#### Web server error log
```
Insert your webserver log here
no log for this issue

#### Nextcloud log (data/nextcloud.log)
no log for this issue

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix #174 

Testing:
1. Create an addressbook with another user in read only mode and share it with your current user.
2. Try to move a contact to this addressbook.

### CODE REVIEW

File: js/services/contact_service.js [hunk 1]
```diff
@@ -251,6 +251,10 @@ angular.module('contactsApp')
 		if (contact.addressBookId === addressbook.displayName) {
 			return;
 		}
+		if(addressbook.readOnly) {
+			OC.Notification.showTemporary(t('contacts', 'You don\'t have permission to write to this addressbook.'));
```
Comment (jonatoni) [inline js/services/contact_service.js]: It should be 'You don't have permission to write to this addressbook.'

Comment (skjnldsv) [inline js/services/contact_service.js]: That's exactly what it's written. 😁

File: js/dav/dav.js [hunk 2]
```diff
@@ -1620,6 +1620,7 @@ var listAddressBooks = _co2['default'].wrap(regeneratorRuntime.mark(function cal
         req = request.propfind({
           props: [{ name: 'displayname', namespace: ns.DAV }, { name: 'owner', namespace: ns.DAV }, { name: 'getctag', namespace: ns.CALENDAR_SERVER }, { name: 'resourcetype', namespace: ns.DAV }, { name: 'sync-token', namespace: ns.DAV },
           //{ name: 'groups', namespace: ns.OC },
+          { name: 'read-only', namespace: ns.OC },
```
Comment (skjnldsv) [inline js/dav/dav.js]: It's the only line I changed on this file, the rest is just Github failing to see I didn't changed anything else.... 😑

File: js/services/contact_service.js (hunk 1 above)
Comment (skjnldsv) [reply js/services/contact_service.js]: If you're talking about the `\'` it's to escape the ' character since we're using it to define the string. :)

**Comment by irgendwie** [comment]
Works and is ok for now - we need to push a new release 👍 
For the future, I'd like to remove read-only addressbooks from the addressbook selection in the details and for contacts of such an addressbook a not editable field.

**Comment by skjnldsv** [comment]
@irgendwie look #294 too :)

File: js/dav/dav.js (hunk 2 above)
Comment (daniellandau) [reply js/dav/dav.js]: This is compiled code. I'm quite sure you shouldn't change it here, but in `js/dav/lib/contacts.js` and then run `make` in `js/dav`.

Comment (skjnldsv) [reply js/dav/dav.js]: Oh you're right! :O

**Pull request merged**